src/Lecture2.hs

```diff
@@ -181,7 +342,14 @@ False
 True
 -}
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
+isIncreasing = go Nothing [] 
+  where go :: Maybe Int -> [Int] -> [Int] -> Bool 
+        go _ _ [] = True
+        go Nothing xs (y:ys) = go (Just y) xs ys
+        go (Just x) xs (y:ys)
+          | x > y = False
+          | otherwise = go (Just y) (xs ++ [x]) ys
```

Comment:
This looks like a correct solution. But it's a bit complicated. Think about how you can implement this function by using `[]` and (`:`) patterns on the input list only without using any extra helper functions and additional helper variables 🤔